src/Middleware/LanguageSwitcherMiddleware.php

```diff
@@ -169,8 +168,8 @@ private function __setCookieAndLocale($locale)
         // @FIXME Should be refactored when cake 3.4 was released
         if (PHP_SAPI !== 'cli') {
             $time = $this->__getCookieExpireTime();
-            I18n::locale($locale);
-            setcookie($this->__getCookieName(), $locale, $time, '/', $this->config('Cookie.domain'));
+            I18n::setLocale($locale);
+            setcookie($this->__getCookieName(), $locale, $time, '/', $this->getConfig('Cookie.domain'));
```

Comment:
We could use https://book.cakephp.org/3.0/en/controllers/request-response.html#setting-cookies to deal with settings the cookie now.